custom_components/amplipi/media_player.py

```diff
@@ -1235,3 +1220,273 @@ async def async_set_volume_level(self, volume):
+                'playing'
+        ):
+            return STATE_PLAYING
+        elif self._current_source.info.state in (
+                'stopped'
+        ):
+            return STATE_IDLE
+        return STATE_IDLE
+
+
+    @property
+    def volume_level(self):
+        """Volume level of the media player (0..1)."""
+        if self._current_source is not None:
+            if len(self._current_zones) > 0:
+                vols = [zone.vol_f for zone in self._current_zones]
+                return sum(vols) / len(vols)
+        return None
+
+    @property
+    def is_volume_muted(self) -> bool:
+        """Boolean if all connected zones are currently muted. If no zones connected, return muted."""
+        if self._current_source is not None:
+            for zone in self._current_zones:
+                if not zone.mute:
+                    return False
+        return True
+ 
+    async def async_select_source(self, source: Optional[str] = None):
+        # This is a home assistant MediaPlayer built-in function, so the source being passed in isn't the same as an amplipi source
+        # the argument "source" can either be the name or entity_id of a zone, group, or amplipi source
+        # As such, this info must be sorted and then sent down the proper logical path
+        if source:
+            if source == "None" and self._current_source is not None:
+                await self._client.set_source(
+                    self._current_source.id,
+                    SourceUpdate(
+                        input='None'
+                    )
+                )
+            elif source == "Any" and self._current_source is None:
+                await self.async_connect_stream_to_source(self._stream)
+            else:
+                amplipi_entity = self.get_entry_by_value(source)
+                if isinstance(amplipi_entity, Source):
+                    await self.async_connect_stream_to_source(self._stream, amplipi_entity)
+                else:
+                    entry = self.get_entry_by_value(source)
+                    if entry:
+                        entry_id = self.extract_amplipi_id_from_unique_id(entry.unique_id)
+                        if isinstance(amplipi_entity, Zone):
+                            await self.async_connect_zones_to_stream(self._stream, [entry_id], None)
+                        elif isinstance(amplipi_entity, Group):
+                            await self.async_connect_zones_to_stream(self._stream, None, [entry_id])
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self._attr_source_list
+
+    @property
+    def extra_state_attributes(self):
+        return self._extra_attributes
+
+    async def _get_extra_attributes(self):
+        if self._current_source is not None:
+            self._extra_attributes = {"amplipi_source_id" : self._current_source.id }
+        else:
+            self._extra_attributes = {"amplipi_source_id" : None }
```

Comment:
This doesn't return anything even though it has get in its name? What is the point here?